diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -171,13 +172,28 @@ controller_interface::return_type DiffDriveController::update(
 
     if (params_.position_feedback)
     {
-      odometry_.update(left_feedback_mean, right_feedback_mean, time);
+      // Warn if actual timestep doesn't match period, 
+      // since 
+      const double measured_period = (
```

Comment:
Why should the measured period be different from the period argument of the update method? This is calculated from the controller_manager, we can assume that this is always valid?

Reply:
I was thinking on this some more myself and came to the same conclusion.

I'm hoping to rework this PR sometime next week focusing on the code quality/cleanup aspects suggested here and remove the inconsistent time handling (which as you suggest is probably a distinction without a difference).

Recently I've switched to using the RT_Linux kernel and set up permissions for real-time and found a fairly dramatic difference in perceived responsiveness with it enabled in the diff drive controller. Something isn't handling time/dates/update timesteps properly and I think I incorrectly identified this as the culprit. It still might be, but I should have started off with unit tests (or at least minimum steps to reproduce) rather than complicating an otherwise straightforward code cleanup.